### ISSUE TITLE
Anwer is total difference

### ISSUE THREAD

**Issue by yujack333** [SongDark/SpectralClustering#1]
![image](https://user-images.githubusercontent.com/26196384/64141476-b749c180-ce3a-11e9-8b2d-e4bd1a3a14ac.png)
i don't change anything. I just download and run. and the anwer is total difference from yours.

**Comment by GuHuangAI** [comment]
The same problem occured in my project，but i solved it by changing the parameter k of fouction myKNN. In my test,the best value is 8 or 9 but the default 10. The reason may be that the algorithm depends on the data's distances. hope can help you.
`adjacent = myKNN(similarity, k=9)`
![image](https://user-images.githubusercontent.com/55237825/71777747-416b8780-2fdf-11ea-835e-d0bb181066db.png)

**Comment by chxyfishge** [comment]
x = zip(x, range(len(x)))
x = sorted(x, key=lambda x:x[0])
H = np.vstack([V[:,i] for (v, i) in x[:500]]).T

Here is a problem, cannot use all eigenVec to calc KMeans. Should use the first n_clusters eigenVec to calc KMeans.
so the code should be:

        x, V = np.linalg.eig(Laplacian)
        dictx = dict(zip(x, range(points_num)))  #not 500, but your points_num
        kEig = np.sort(x)[0:self.n_clusters]   #self.n_clusters is count of clusters
        ix = [dictx[k] for k in kEig]
        H = V[:,ix]
        sp_kmeans = KMeans(n_clusters=2).fit(H)

then it will be not sensitive to k of KNN.

**Comment by GuHuangAI** [comment]
> x = zip(x, range(len(x)))
> x = sorted(x, key=lambda x:x[0])
> H = np.vstack([V[:,i] for (v, i) in x[:500]]).T
> 
> Here is a problem, cannot use all eigenVec to calc KMeans. Should use the first n_clusters eigenVec to calc KMeans.
> so the code should be:
> 
> ```
>     x, V = np.linalg.eig(Laplacian)
>     dictx = dict(zip(x, range(points_num)))  #not 500, but your points_num
>     kEig = np.sort(x)[0:self.n_clusters]   #self.n_clusters is count of clusters
>     ix = [dictx[k] for k in kEig]
>     H = V[:,ix]
>     sp_kmeans = KMeans(n_clusters=2).fit(H)
> ```
> 
> then it will be not sensitive to k of KNN.

yeah, it's useful. but different num_points also cause different results.
